src/main/java/net/minecraftforge/fml/ModLoader.java

```diff
@@ -198,6 +198,8 @@ public void gatherAndInitializeMods(final ModWorkManager.DrivenExecutor syncExec
         CapabilityManager.INSTANCE.injectCapabilities(modList.getAllScanData());
         statusConsumer.ifPresent(c->c.accept("Populating registries"));
         dispatchAndHandleError(ModLoadingStage.LOAD_REGISTRIES, syncExecutor, parallelExecutor, periodicTask);
+        statusConsumer.ifPresent(c->c.accept("Adding custom tag types"));
+        GameData.setCustomTagTypesFromRegistries();
```

Comment:
Since this is not trying to call mod supplied code, but is rather, just parsing out metadata from the registries into structure, this is the best solution.